test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/IntegerGenerator.java

```diff
@@ -1,15 +1,64 @@
 package uk.gov.justice.services.test.utils.core.random;
 
+import static java.lang.String.format;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
 public class IntegerGenerator extends Generator<Integer> {
 
-    private final Integer max;
+    /**
+     * The minimum value of the range
+     */
+    private final int min;
 
-    public IntegerGenerator(final Integer max) {
+    /**
+     * The maximum value of the range
+     */
+    private final int max;
+
+    /**
+     * Package Access only
+     * 
+     * @see RandomGenerator
+     */
+    IntegerGenerator() {
+        this.min = Integer.MIN_VALUE;
+        this.max = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Package Access only
+     * 
+     * @param max value excluded
+     * @see RandomGenerator
+     */
+    IntegerGenerator(final int max) {
+        this.min = 0;
+        this.max = max;
+    }
+    /**
+     * Package Access only
+     * 
+     * @param min value included
+     * @param max value excluded
+     * @see RandomGenerator
+     */
+    IntegerGenerator(final int min, final int max) {
+        if (min >= max) {
+            throw new IllegalArgumentException(
+                            format("%s cannot be greater or equal to %s", min, max));
```

Comment:
Could have better error message? 
E.g.
Min value cannot be greater than or equal to Max value, got Min: %s and Max: %s